bin/spack.ps1

```diff
@@ -79,7 +86,7 @@ function Invoke-SpackEnv {
                 }
                 else {
                     $SpackEnv = $(python $Env:SPACK_ROOT/bin/spack $SpackCMD_params env activate "--pwsh" $SubCommandSubCommandArgs)
-                    $ExecutionContext.InvokeCommand($SpackEnv)
+                    Set-SpackEnv $SpackEnv
```

Comment:
What is the difference between this and calling

```
Invoke-Expression $Spackenv
```

here? In other words, I'm not clear on the significance of the `for` loop in `Set-SpackEnv`.

Reply:
Because `$Spackenv` is not an object that `Invoke-Expression` understands. `Invoke-Expression` attempts to interpret a string as a powershell operation and invoke it, however, `$Spackenv` is a `System.Object[]` (powershell list object) and is not something that `Invoke-Expression` can understand as a command. 
Additionally, just concatenating all the list items together into a single string delineated by spaces or newlines is also not a legal invocation of `Invoke-Expression` as it tries to interpret the entire string as a single operation, which of course, it's not. So we loop through all the returned values. 

As to why `$spackenv` is a list in the first place: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators?view=powershell-7.3#subexpression-operator--
From the docs: multiple results are returned as an array.

Reply:
I've added a comment elaborating on this